src/order/locally_finite.lean

```diff
@@ -601,9 +745,21 @@ instance : locally_finite_order (subtype p) :=
     subtype.coe_lt_coe],
   finset_mem_Ioo := λ a b x, by simp_rw [finset.mem_subtype, mem_Ioo, subtype.coe_lt_coe] }
 
-variables (a b : subtype p)
+instance [locally_finite_order_top α] : locally_finite_order_top (subtype p) :=
+{ finset_Ici := λ a, (Ici (a : α)).subtype p,
+  finset_Ioi := λ a, (Ioi (a : α)).subtype p,
+  finset_mem_Ici := λ a x, by simp_rw [finset.mem_subtype, mem_Ici, subtype.coe_le_coe],
+  finset_mem_Ioi := λ a x, by simp_rw [finset.mem_subtype, mem_Ioi, subtype.coe_lt_coe] }
+
+instance [locally_finite_order_bot α] : locally_finite_order_bot (subtype p) :=
+{ finset_Iic := λ a, (Iic (a : α)).subtype p,
+  finset_Iio := λ a, (Iio (a : α)).subtype p,
+  finset_mem_Iic := λ a x, by simp_rw [finset.mem_subtype, mem_Iic, subtype.coe_le_coe],
+  finset_mem_Iio := λ a x, by simp_rw [finset.mem_subtype, mem_Iio, subtype.coe_lt_coe] }
```

Comment:
Does this form a diamond with the `is_empty` instance when `p = false`?

Reply:
(not a deal-breaker, but we should add a comment about it)

Reply:
In fact, we should add the subsingleton instance anyway to match the one for `locally_finite_order`.

Reply:
```
example {α : Type*} [preorder α] [locally_finite_order_top α] :
  @subtype.locally_finite_order_top α _ (λ _, false) _ _ = is_empty.to_locally_finite_order_top :=
rfl -- fails
```
And indeed I don't think we care much about empty subtypes.

Reply:
This suggests to me that `is_empty.to_locally_finite_order_top` should not be an instance. I would guess it creates diamonds for `fin 0 × fin 1` too

Reply:
I would guess it's not actually useful as an instance anywhere anyway (except on pempty, which you can just add an instance for)